test/modules/obsagent-client/src/main.rs

```diff
@@ -0,0 +1,204 @@
+                .long("batching-enabled")
+                .help("Enables or disables batch recording of measurements.")
+        )
+        .arg(
+            Arg::with_name("otlp-endpoint")
+                .short("e")
+                .long("otlp-endpoint")
+                .help("Endpoint to which OTLP messages will be sent.")
+        )
+        .arg(
+            Arg::with_name("enable-value-recorder")
+                .short("r")
+                .long("enable-value-recorder")
+                .help("Enables or disables value recorder instrument.")
+        )
+        .arg(
+            Arg::with_name("enable-value-observer")
+                .short("v")
+                .long("enable-value-observer")
+                .help("Enables or disables value observer instrument.")
+        )
+        .get_matches();
+
+    let config = Config {
+        update_rate: if let Ok(v) = std::env::var("OBS_AGENT_CLIENT_UPDATE_RATE") {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("update-rate"), f64).unwrap_or(1.0)
+        },
+        push_rate: if let Ok(v) = std::env::var("OBS_AGENT_CLIENT_PUSH_RATE") {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("push-rate"), f64).unwrap_or(0.2)
+        },
+        batching_enabled: if let Ok(v) = std::env::var("OBS_AGENT_CLIENT_BATCHING_ENABLED") {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("batching-enabled"), bool).unwrap_or(false)
+        },
+        otlp_endpoint: if let Ok(v) = std::env::var("OBS_AGENT_OTLP_ENDPOINT") {
+            v.parse()?
+        } else {
+            matches
+                .value_of("otlp-endpoint")
+                .unwrap_or("http://localhost:4317")
+                .to_string()
+        },
+        enable_value_recorder: if let Ok(v) =
+            std::env::var("OBS_AGENT_CLIENT_ENABLE_VALUE_RECORDER")
+        {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("enable-value-recorder"), bool).unwrap_or(false)
+        },
+        enable_value_observer: if let Ok(v) =
+            std::env::var("OBS_AGENT_CLIENT_ENABLE_VALUE_OBSERVER")
+        {
+            v.parse()?
+        } else {
+            value_t!(matches.value_of("enable-value-observer"), bool).unwrap_or(false)
+        },
+    };
+    println!("{:?}", config);
+    Ok(config)
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+    let config = init_config()?;
+    let _started = init_meter(1.0 / config.push_rate, config.otlp_endpoint)?;
+    let meter = global::meter("microsoft.com/azureiot-edge");
+
+    // Init synchronous instruments
+    let counter = meter.u64_counter("u64_counter_example").init();
+    let ud_counter = meter
+        .i64_up_down_counter("i64_up_down_counter_example")
+        .init();
+    let mut value_recorder: Option<ValueRecorder<f64>> = None;
+    if config.enable_value_recorder {
```

Comment:
Why is this configurable?

Reply:
I made it configurable because the memory usage of the ValueRecorder instrument was growing continuously because it was  holding on to every measurement.  For this reason, I wanted the ability to disable the ValueRecorder for some tests. I looked into it though and it looks like the unbounded memory growth was because I was using the `Exact` aggregator selector: https://docs.rs/opentelemetry/0.13.0/opentelemetry/sdk/metrics/selectors/simple/enum.Selector.html#variant.Exact

I switched to the Histogram selector, which better aligns with the behavior we're looking for, so I can now remove the enable params.